### DISCUSSION TITLE
Don't block entering SONG or CLIP when any menu is open/active.

### DISCUSSION THREAD

**Discussion by ok-reza** [SynthstromAudible/DelugeFirmware#2774] [Ideas]
### Please describe the problem:

If you are in a clip and have a parameter menu open (eg. lpf frequency) you are unable to press SONG to quickly exit to SONG view. This occurs for any menu type being open.

### What is the expected behavior?

Allow using the SONG and CLIP pads to exit to those views while a menu is open.

When performing live, it's possible you enter a clip and adjust a parameter live, say ARP RATE. After you've changed that value, while that menu is open, pressing SONG has no effect and creates a block to quickly get back to SONG view to quickly mute/launch the necessary clips.

If SONG or CLIP is pressed while a menu is open, I think it should simply exit the menu while transitioning views.

### What hardware did you reproduce it with?

OLED

### What firmware did you reproduce it with?

Beta

### What is the firmware name:

1.2 beta

**Comment by duncanrmi** [comment]
this bugs the hell out of me too.
I think the rationale behind it is to prompt you to save the synth (or whatever) you've just been tweaking, & this raises a larger question over what should be the expected behaviour when editing synths & kits while in the throes of arranging a song or jamming.
personally, I'd like to see a dot appear next to the synth/kit name, to tell you it isn't the same as the remembered one.

(& yes, if you happen to be in an edit mode, & 'back' is flashing at you, you should be free to ignore it & drop back into song view!)

THEN when you go to save the song, you would have an additional choice to make- auto-save all the edited synths/kits with incremented numbers & have the song reference them rather than the original synths/kits you started with, or put one/all of them back to how they were. 
this may seem tedious, but now the work is happening when you go to save the song, & not during your session.

**Comment by RJ-Eckie** [comment]
100%! Current implementation does not make sense to me

**Comment by bukru** [comment]
I agree 100%
